content/posts/manual-uid-updates-post-migration.md

```diff
@@ -0,0 +1,171 @@
+
+```yaml
+  "openshift.io/sa.scc.mcs": "s0:c26,c15",
+  "openshift.io/sa.scc.supplemental-groups": "1000680000/10000",
+  "openshift.io/sa.scc.uid-range": "1000680000/10000"
+```
+
+> It is recommended to keep the original values backed up so that the workloads can be rolled back in case of failures in further steps.
+
+#### Step 2: Quiescing apps
+
+After step 1, the Namespaces will have new non-overlapping UID ranges assigned to them. However, the workloads will still be running with their old UIDs. The new UIDs will take effect only after the workloads are restarted. In this step, the workloads will simply be quiesced down.
+
+Before quiescing down, the current replica values of workload resources need to stored. The value will be needed at the time of un-quiescing the workloads. They can be simply stored as annotation on the workload itself. An example of saving replica value for a Deployment resource:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    previous-replicas: "3"   <--- Saving replica count
+  [...]
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 3
+  [...]
+```
+
+Once the annotations are applied, the replicas of workload resources will be set to 0:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    previous-replicas: "3"
+  [...]
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 0   <--- Quiesce down
+  [...]
+```
+
+#### Step 3: Updating ownership of PersistentVolume data
+
+In this step, the file ownership of the data will be updated manually. Note that this step may or may not be required depending on the underlying storage used by the PVCs.
+#### When this step may not be required?
+
+If the storage is a block storage, Kubernetes will automatically `chown` files on the volumes when the workloads are un-quiesced in Step 4. AWS EBS, GCE Persistent Disks are some examples of block storages that Kubernetes automatically updates the ownership of the data based on the UID of running workloads. One way to determine whether or not the automatic `chown` is supported is by looking at the `fsType` attribute of the underlying PersistentVolume object:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: pvc-227d5669-01fc-49df-8d34-294da0d9ee60
+spec:
+  gcePersistentDisk:
+    fsType: ext4    <--- fsType field is set
+```
+
+If this field is set, Kubernetes will automatically `chown` the volume. 
+
+If the storage is a shared storage, an additional list of Supplemental Groups can be provided to the workloads. The workloads can simply access the files using one of the GIDs provided in the Supplemental Groups section of the Pod's security context: 
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: restic-zppwk
+  namespace: openshift-migration
+spec:
+  [...]
+  securityContext:
+    supplementalGroups:
+    - 1000
```

Comment:
This might require a SCC change
```
To remedy the situation in the previous example, a custom SCC can be created such that:

a minimum and max group ID are defined,

ID range checking is enforced, and

the group ID of 5555 is allowed.
```
https://docs.openshift.com/container-platform/3.11/install_config/persistent_storage/pod_security_context.html#supplemental-groups may be find a 4.x doc link it here?

Reply:
+1 not looking for detail just a link to openshift doc will do I think.